### PR TITLE
SPE-2003: Add medical outcome deviation audit

### PR DESCRIPTION
**Pull request by JamesJedi420**
## Summary

- Adds pure deterministic `buildMedicalOutcomeDeviationAuditReport` for predicted-vs-observed medical/containment outcome deviation (SPE-2003).
- Linear: https://linear.app/spectranoir/issue/SPE-2003/batch-9977-medical-outcome-deviation-audit-system
- SPE-2073 remains Duplicate/canceled; **SPE-2003 is the canonical implementation owner**.

## Smallest boundary

- New domain helper + fixture tests only.
- No GameState, weekly tick, UI, SPE-2010 import, staff alignment, scorecard, blame routing, doctrine, or policy selector.

## Files changed

- `src/domain/medicalOutcomeDeviationAudit.ts`
- `src/test/medicalOutcomeDeviationAudit.test.ts`

## Review hardening (43b0336)

- Honor `treatMissingExpectedEscalationAsZero: false` (skip escalation comparison when expected count omitted).
- Match escalation events to paired observation week on fallback pairing.
- Pre-group observations by subject/protocol; single-pass escalation event accounting.
- Simplify governance notification level resolution; document threshold normalization.

## Validation

- `npm run test:run -- src/test/medicalOutcomeDeviationAudit.test.ts` (31 tests)
- `npm run test:run -- src/test/staffTreatmentTelemetry.test.ts` (adjacent)
- `npm run test:run -- src/test/uncertainWorldState.test.ts` (adjacent)
- `npm run test:run` (3119 tests)
- `npm run lint`
- `git diff --check`

## Gap / edge-case / boundary audit

| Category | Result |
| --- | --- |
| Audit model | Expected/observed independent; escalation option honored; governance candidates have no side effects |
| Pairing | subject+protocol+week with fallback; observation week used for escalation events; dedupe first-wins |
| Determinism | Stable sort, lines, summary; inputs not mutated |
| Boundaries | No GameState, SPE-2010, UI, notifications, batch strings in lines |
| Docs | Module header + option JSDoc for threshold normalization |

## Out of scope

- GameState persistence, runtime integration, UI
- SPE-2010 adapter, recovery/trauma projection, uncertainWorldState
- SPE-2008 scorecard, SPE-2006 blame routing, SPE-2001 doctrine, SPE-2004 policy selector

## Test plan

- [x] Targeted SPE-2003 tests (31)
- [x] Adjacent staff telemetry + uncertain world state tests
- [x] Full suite + lint

### CODE REVIEW

**Comment by JamesJedi420** [comment]
Review hardening pushed in 43b0336.

**Blocking fix:** \	reatMissingExpectedEscalationAsZero: false\ now skips escalation-excess comparison and governance triggers that depend only on skipped escalation.

**Also addressed:**
- Escalation events match paired observation week on fallback (not expectation week).
- Observations pre-grouped by subject/protocol; single pass tracks used escalation event IDs.
- Governance level resolver simplified to directive → governance → site_lead.
- Threshold normalization documented on options/module header.

**Validation:** 31 SPE-2003 tests, full suite 3119, lint clean.

File: src/domain/medicalOutcomeDeviationAudit.ts [hunk 1]
```diff
@@ -0,0 +1,784 @@
...
+}
+
+function resolveExpectedEscalationCount(
+  expectation: MedicalExpectedOutcome,
+  options: Required<MedicalOutcomeDeviationAuditOptions>
+): number {
+  if (expectation.expectedEscalationCount !== undefined) {
+    return expectation.expectedEscalationCount
+  }
+  return options.treatMissingExpectedEscalationAsZero ? 0 : 0
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/medicalOutcomeDeviationAudit.ts]: Fixed in 43b0336: \
esolveExpectedEscalation\ returns \compareEscalation: false\ when the option is disabled and expected count is omitted, skipping escalation findings and escalation-only governance candidates. Regression tests added.

[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/medicalOutcomeDeviationAudit.ts]: Fixed in 43b0336: \
esolveExpectedEscalation\ returns \compareEscalation: false\ when the option is disabled and expected count is omitted, skipping escalation findings and escalation-only governance candidates. Regression tests added.

[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/medicalOutcomeDeviationAudit.ts]: Fixed in 43b0336: \
esolveExpectedEscalation\ returns \compareEscalation: false\ when the option is disabled and expected count is omitted, skipping escalation findings and escalation-only governance candidates. Regression tests added.

File: src/domain/medicalOutcomeDeviationAudit.ts [hunk 2]
```diff
@@ -0,0 +1,784 @@
...
+}
+
+function matchEscalationEvents(input: {
+  expectation: MedicalExpectedOutcome
+  observation: MedicalObservedOutcome | undefined
+  events: readonly MedicalOutcomeEscalationEvent[]
+}): MedicalOutcomeEscalationEvent[] {
+  const { expectation, observation, events } = input
+  const week = expectation.expectedByWeek ?? observation?.observedWeek
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/medicalOutcomeDeviationAudit.ts]: Fixed in 43b0336: \
esolveEventMatchWeek\ prefers \observation.observedWeek\ over \expectation.expectedByWeek\ so fallback-paired observations match week-specific escalation events. Test added for expected week 3 / observation week 4.

File: src/domain/medicalOutcomeDeviationAudit.ts [hunk 3]
```diff
@@ -0,0 +1,784 @@
...
+  return worst
+}
+
+function resolvePairedObservation(
+  expectation: MedicalExpectedOutcome,
+  observations: readonly MedicalObservedOutcome[]
+): MedicalObservedOutcome | undefined {
+  const forSubjectProtocol = observations.filter((observation) =>
+    matchesSubjectProtocol(observation, expectation)
+  )
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/medicalOutcomeDeviationAudit.ts]: Fixed in 43b0336: observations are pre-grouped by subject/protocol; escalation event usage is tracked during the main expectation loop (no second pairing pass).

File: src/domain/medicalOutcomeDeviationAudit.ts [hunk 4]
```diff
@@ -0,0 +1,784 @@
...
+    return 'governance'
+  }
+  if (
+    input.hasCriticalSibling ||
+    outcomeGap >= options.criticalOutcomeGapThreshold ||
+    hasOutcomeDeviation
+  ) {
+    return 'site_lead'
+  }
+  return 'site_lead'
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/medicalOutcomeDeviationAudit.ts]: Fixed in 43b0336: governance resolver is directive → governance → site_lead with a single fallback.

File: src/domain/medicalOutcomeDeviationAudit.ts [hunk 5]
```diff
@@ -0,0 +1,784 @@
...
+  const usedEscalationEventIds = new Set<string>()
+  for (const expectation of expectedOutcomes) {
+    const observation = resolvePairedObservation(expectation, observedOutcomes)
+    for (const event of matchEscalationEvents({ expectation, observation, events: escalationEvents })) {
+      usedEscalationEventIds.add(event.eventId)
+    }
+  }
+  const unpairedEscalationEventCount = escalationEvents.filter(
+    (event) => !usedEscalationEventIds.has(event.eventId)
+  ).length
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/medicalOutcomeDeviationAudit.ts]: Fixed in 43b0336: observations are pre-grouped by subject/protocol; escalation event usage is tracked during the main expectation loop (no second pairing pass).

File: src/domain/medicalOutcomeDeviationAudit.ts [hunk 6]
```diff
@@ -0,0 +1,784 @@
...
+      DEFAULT_OPTIONS.criticalEscalationExcess
+    ),
+    governanceEscalationExcessThreshold: normalizeThreshold(
+      options?.governanceEscalationExcessThreshold,
+      DEFAULT_OPTIONS.governanceEscalationExcessThreshold
+    ),
+    treatMissingExpectedEscalationAsZero:
+      options?.treatMissingExpectedEscalationAsZero ??
+      DEFAULT_OPTIONS.treatMissingExpectedEscalationAsZero,
+  }
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/medicalOutcomeDeviationAudit.ts:228]: Addressed in 43b0336: module header and option JSDoc note that thresholds/margins normalize to finite non-negative integers while symptom burden deltas remain signed on inputs.

File: src/domain/medicalOutcomeDeviationAudit.ts [hunk 7]
```diff
@@ -0,0 +1,784 @@
...
+  if (hasEscalationExcess || (hasMaterialGap && hasSymptomWorsened)) {
+    return 'governance'
+  }
+  if (
+    input.hasCriticalSibling ||
+    outcomeGap >= options.criticalOutcomeGapThreshold ||
+    hasOutcomeDeviation
+  ) {
+    return 'site_lead'
+  }
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/medicalOutcomeDeviationAudit.ts]: Fixed in 43b0336: governance resolver is directive → governance → site_lead with a single fallback.

File: src/domain/medicalOutcomeDeviationAudit.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/medicalOutcomeDeviationAudit.ts]: Fixed in 43b0336: observations are pre-grouped by subject/protocol; escalation event usage is tracked during the main expectation loop (no second pairing pass).
